### ISSUE TITLE
Check output filename extensions before writing

### ISSUE THREAD

**Issue by cjordan** [MWATelescope/Birli#41]
I did something a little silly, but it exposed a poor error message:

```bash
# 1090008640.ms
birli *gpubox* -m *.metafits -u 1090008640.ms
# prints out
thread 'main' panicked at 'couldn't initialise uvfits writer: IO(Os { code: 21, kind: IsADirectory, message: "Is a directory" })', src/main.rs:1036:14
```

(I'm using the v0.5.0 branch and it failed on the line containing `.expect("couldn't initialise uvfits writer")`)

This error happens because the "file" already exists, and Rust notes that this is a directory when trying to initialise the output as a file.  It would be good to have existing files checked and handled in this case, but also user input filenames checked; .ms files write as MS, .uvfits write as uvfits, etc.

**Comment by d3v-null** [comment]
I agree that this would be handy, and we could probably copy the code from Hyperdrive that does this.

At least the error message is a bit nicer since Marlu 0.6.0 https://github.com/MWATelescope/Marlu/commit/01d2393b302af73a5c7ae6be7bab9be5586e75b3

```
cannot create directory, path={path} already exists and is not a directory
```
